packages/nextjs/src/config/webpack.ts

```diff
@@ -195,37 +194,41 @@ function addFileToExistingEntryPoint(
   filepath: string,
 ): void {
   // can be a string, array of strings, or object whose `import` property is one of those two
-  let injectedInto = entryProperty[entryPointName];
-
-  // Sometimes especially for older next.js versions it happens we don't have an entry point
-  if (!injectedInto) {
-    // eslint-disable-next-line no-console
-    console.error(`[Sentry] Can't inject ${filepath}, no entrypoint is defined.`);
-    return;
-  }
+  const currentEntryPoint = entryProperty[entryPointName];
+  let newEntryPoint: EntryPointValue;
 
   // We inject the user's client config file after the existing code so that the config file has access to
   // `publicRuntimeConfig`. See https://github.com/getsentry/sentry-javascript/issues/3485
-  if (typeof injectedInto === 'string') {
-    injectedInto = [injectedInto, filepath];
-  } else if (Array.isArray(injectedInto)) {
-    injectedInto = [...injectedInto, filepath];
-  } else {
-    let importVal: string | string[];
+  if (typeof currentEntryPoint === 'string') {
+    newEntryPoint = [currentEntryPoint, filepath];
+  } else if (Array.isArray(currentEntryPoint)) {
+    newEntryPoint = [...currentEntryPoint, filepath];
+  }
+  // descriptor object (webpack 5+)
+  else if (typeof currentEntryPoint === 'object' && 'import' in currentEntryPoint) {
+    const currentImportValue = currentEntryPoint.import;
+    let newImportValue: string | string[];
 
-    if (typeof injectedInto.import === 'string') {
-      importVal = [injectedInto.import, filepath];
+    if (typeof currentImportValue === 'string') {
+      newImportValue = [currentImportValue, filepath];
     } else {
-      importVal = [...injectedInto.import, filepath];
+      newImportValue = [...currentImportValue, filepath];
     }
 
-    injectedInto = {
-      ...injectedInto,
-      import: importVal,
+    newEntryPoint = {
+      ...currentEntryPoint,
+      import: newImportValue,
     };
+  } else {
+    logger.warn(
+      `Could not inject SDK initialization code into entry point ${entryPointName}, as it is not a recognized format.\n`,
+      `Expected: string | Array<string> | { [key:string]: any, import: string | Array<string> }\n`,
+      `Got: ${currentEntryPoint}`,
```

Comment:
Tbh I wouldn't even mind throwing an error to fail fast, but more than happy with a warning.

Reply:
I switched it to logging an error rather than a warning - won't break the build, but more likely to be seen.